src/layout/IHyprLayout.cpp

```diff
@@ -396,6 +396,139 @@ void IHyprLayout::onEndDragWindow() {
     g_pInputManager->m_bWasDraggingWindow = false;
 }
 
+inline bool canSnap(const double sideA, const double sideB, const double gap) {
+    return std::abs(sideA - sideB) < gap;
+}
+
+void snapMoveLeft(double& pos, double& len, const double p) {
+    pos = p;
+}
+void snapMoveRight(double& pos, double& len, const double p) {
+    pos = p - len;
+}
+
+void snapResizeLeft(double& pos, double& len, const double p) {
+    len += pos - p;
+    pos = p;
+}
+void snapResizeRight(double& pos, double& len, const double p) {
+    len = p - pos;
+}
+
+typedef void (*SnapFn)(double& pos, double& len, const double p);
+
+void IHyprLayout::performSnap(Vector2D& pos, Vector2D& size, PHLWINDOW DRAGGINGWINDOW, const eMouseBindMode mode) {
+    static auto SNAPWINDOWGAP  = CConfigValue<Hyprlang::INT>("general:snap:window_gap");
+    static auto SNAPMONITORGAP = CConfigValue<Hyprlang::INT>("general:snap:monitor_gap");
+
+    int         corner, snaps = 0;
+    SnapFn      snapUp, snapDown, snapLeft, snapRight;
```

Comment:
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

Reply:
I declare the variables, and then define them in the if-else clause that immediately follows. That way, only 1 condition needs to be checked to assign multiple variables. They'll never be undefined when actually being called upon.